zconf-ng.h.in

```diff
@@ -48,30 +55,29 @@
 /* Type declarations */
 
 
-#if defined(_WIN32)
-   /* If building or using zlib as a DLL, define ZLIB_DLL.
-    * This is not mandatory, but it offers a little performance increase.
-    */
-#  if defined(ZLIB_DLL) && (!defined(__BORLANDC__) || (__BORLANDC__ >= 0x500))
-#    ifdef ZLIB_INTERNAL
-#      define ZEXTERN extern __declspec(dllexport)
-#    else
-#      define ZEXTERN extern __declspec(dllimport)
-#    endif
-#  endif  /* ZLIB_DLL */
-   /* If building or using zlib with the WINAPI/WINAPIV calling convention,
-    * define ZLIB_WINAPI.
-    * Caution: the standard ZLIB1.DLL is NOT compiled using ZLIB_WINAPI.
-    */
-#  ifdef ZLIB_WINAPI
-#    include <windows.h>
-     /* No need for _export, use ZLIB.DEF instead. */
-     /* For complete Windows compatibility, use WINAPI, not __stdcall. */
-#    define ZEXPORT WINAPI
-#    define ZEXPORTVA WINAPIV
+/* If building or using zlib as a DLL, define ZLIB_DLL.
+ * This is not mandatory, but it offers a little performance increase.
+ */
+#if defined(ZLIB_DLL) && (defined(_WIN32) || (__has_declspec_attribute(dllexport) && __has_declspec_attribute(dllimport)))
```

Comment:
Can you move `defined(_WIN32)` in both statements back to where it was? It makes it harder to read the diff and it makes this line in particular long.

Reply:
Just want to make sure we're on the same page here. So the background is that LLVM allows a platform to support MS `__declspec` extensions. This is for sure used with clang on Windows but it doesn't have to be a Windows target. In my case this is an adjustment for compiling for PlayStation.

Because of this I split the setting of `ZEXTERN` and `ZLIB_WINAPI` into separate sections. The former after this is not Windows specific while the later will always be Windows specific.

Reply:
I see ok.